azuredevops/internal/service/approvalsandchecks/common_check.go

```diff
@@ -101,6 +105,7 @@ func doBaseExpansion(d *schema.ResourceData, checkType *pipelineschecksextras.Ch
 			Id:   converter.String(d.Get("target_resource_id").(string)),
 			Type: converter.String(d.Get("target_resource_type").(string)),
 		},
+		Version: converter.Int(d.Get("version").(int)),
```

Comment:
Since `version` is computed we don't need to send it to the service

Reply:
@xuzhang3 When I remove that line of code, the API has the same error.

![image](https://github.com/microsoft/terraform-provider-azuredevops/assets/10904682/c074abc1-3899-4311-adc2-8dd109bbaec9)


On a create, the API doesn't require a `version` property in the payload, but it returns a `version` property.
On an update, the API requires a `version` property, which is a whole integer which increments each time the resources is updated.